### DISCUSSION TITLE
Quick Fixes Filter Prevents "Stop-Video Autoplay" from Working

### DISCUSSION THREAD

**Discussion by Readmit0** [uBlockOrigin/uAssets#19892] [General]
### Prerequisites

- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] I am running the latest version of uBlock Origin (uBO).

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

By following this tutorial on YouTube, I am able to prevent YouTube and other media websites from autoplaying videos when I click on them.

https://www.youtube.com/watch?v=6xnvbc8oh0s

Now, in order to get a video to play, I have to click on the video, which is what I want. I don't want videos to start playing until I decide to click on them. The problem is that Ublock Origin's Quick Fixes filter prevents this from working, so that videos will autoplay regardless of whether I follow the instructions in this tutorial or not. I am reporting this to you to let you know that the Quick Fixes filter needs to be changed, so that this fix won't be blocked by it. This was a brand new Windows 10 installation with a brand new untouched Edge browser, with Ublock Origin being the one extension used on it. Please watch this video to see the change that I made to Edge in order to get videos to not autoplay.

### URL(s) where the issue occurs.

youtube.com
dailymotion.com

### uBO version

uBlock Origin 1.52.2

### Browser name and version

Edge Version 117.0.2045.41

### Settings

I unchecked the hot fixes so that the steps that I followed in the tutorial to prevent autoplay on videos would work. Leaving it checked causes it not to work.

**Comment by stephenhawk8054** [comment]
Hmmm... For me it's not related to quick fix list, I have to turn off all scriptlets and all cosmetics filters in order for that flag to have effects.

Steps to reproduce:

1. Open Edge, go to `edge://flags/`, enable `Show block option in autoplay settings` and restart the browser
2. Go to `edge://settings/content/mediaAutoplay` -> Choose `Block`
3. Go to `https://www.youtube.com/`
4. Right click on any video -> Open in new tab
5. In new tab, the video autoplays

Now add

```adb
youtube.com#@#+js()
@@||youtube.com^$ehide
```

and repeat step 3 -> 5, the video autoplay stops.

---

I'm not able to investigate more.

@gorhill Sorry, if possible, can you take a look?

**Comment by Readmit0** [comment]
I've done the steps you mentioned. They were in the video that I linked to. So long as the Quick Fix check box is checked in Ublock origin's filter list, the videos will autoplay. If I uncheck it, the videos won't play until I click on them, which is what I want.

**Comment by stephenhawk8054** [comment]
As I said, I cannot make it auto-stop unless I add the 2 filters above

**Comment by Readmit0** [comment]
> As I said, I cannot make it auto-stop unless I add the 2 filters above

My mistake. I didn't see the part where you mentioned the custom filters. I put them in the MyFilters section of Ublock Origin, and it seems to work. But yeah, you have to open them in a new tab for it to work.

**Comment by stephenhawk8054** [comment]
Those filters will introduce ads so don't use them. They are just for testing.

> you have to open them in a new tab for it to work

You mean there are other scenarios of auto-stop?

**Comment by Readmit0** [comment]
> Those filters will introduce ads so don't use them. They are just for testing.
> 
> > you have to open them in a new tab for it to work
> 
> You mean there are other scenarios of auto-stop?

Yeah, my mistake, it doesn't work with those filters anyway. Do you think this can be resolved somehow?

**Comment by stephenhawk8054** [comment]
> Do you think this can be resolved somehow?

Wait for others to help investigate. I can't answer that.

> I'm not able to investigate more.

**Comment by stephenhawk8054** [comment]
@Readmit0 Anyway, please describe the steps to reproduce your issue in much details as possible so others can know how to reproduce it. 

I still don't know in which scenarios you mean by

> it doesn't work with those filters anyway

**Comment by Readmit0** [comment]
> @Readmit0 Anyway, please describe the steps to reproduce your issue in much details as possible so others can know how to reproduce it.
> 
> I still don't know in which scenarios you mean by
> 
> > it doesn't work with those filters anyway

I've described the issue. I'm not sure how much clearer I can make it. On YouTube, I have the choice of either disabling Ublock Origin just for YouTube, or I have the choice to uncheck mark the Quick Fix Filters option. If I don't do this, I won't be able to get stop the autoplay on new tabs for videos.

**Comment by stephenhawk8054** [comment]
> autoplay on new tabs for videos.

By this you mean the step 3 and 4 in https://github.com/uBlockOrigin/uAssets/issues/19861#issuecomment-1734639611 ?

**Comment by Readmit0** [comment]
> > autoplay on new tabs for videos.
> 
> By this you mean the step 3 and 4 in [#19861 (comment)](https://github.com/uBlockOrigin/uAssets/issues/19861#issuecomment-1734639611) ?

Yes.

**Comment by stephenhawk8054** [comment]
That's what I mean by **STEPS** to reproduce. I was not sure if you were implying any other scenarios with different steps or you were doing the same.

Anyway, the videos are auto-stop correctly for me with those 2 filters. Video recording: https://github.com/uBlockOrigin/uAssets/assets/66517106/b8eb7448-3632-4623-89fc-555d7e304cb6

Let's wait for other investigations

**Comment by Readmit0** [comment]
> That's what I mean by **STEPS** to reproduce. I was not sure if you were implying any other scenarios with different steps or you were doing the same.
> 
> Anyway, the videos are auto-stop correctly for me with those 2 filters. Video recording: https://github.com/uBlockOrigin/uAssets/assets/66517106/b8eb7448-3632-4623-89fc-555d7e304cb6
> 
> Let's wait for other investigations

Those 2 filters worked and then they stopped working for some reason.

**Comment by Readmit0** [comment]
> That's what I mean by **STEPS** to reproduce. I was not sure if you were implying any other scenarios with different steps or you were doing the same.
> 
> Anyway, the videos are auto-stop correctly for me with those 2 filters. Video recording: https://github.com/uBlockOrigin/uAssets/assets/66517106/b8eb7448-3632-4623-89fc-555d7e304cb6
> 
> Let's wait for other investigations

We're still waiting for gorhill to respond or look into it, right?

**Comment by stephenhawk8054** [comment]
Depends on how any others can reproduce. The issue should be moved to https://github.com/uBlockOrigin/uBlock-issues IMO, this does not seem like filter list issue.

---

cc @gwarser @uBlock-user I'm not sure which repo it should be.

**Comment by uBlock-user** [comment]
None of the filters in the list have anything to do with autoplay, so something else is at play here.

**Comment by ghost** [comment]
I have no idea. Looks a bit too random to me. Must be some racing condition.

**Comment by garry-ut99** [comment]
The first thing is that Block Media Edge's feature works only for videos opened in a new tab, that's just the way it works, so users should not expect the feature to work when opening videos in the same tab, and should not consider it being an issue (at least not uBO's)

As for the videos opened in a new tab, yes - me too can reproduce the issue with uBO enabled :
   - on the both mentioned websites (youtube and dailymotion)
   - the exception filters provided by _stephenhawk8054_ for youtube work for me too (didn't try with dailymotion)
   - disabling only "Quick fixes" filter list didn't work for me either, but if additionally disabled "Ads" filter list, it does work
   - also one more uBO user (ms4132) reported the same issue almost 2 years ago in this thread : 
   `https://techcommunity.microsoft.com/t5/discussions/media-autoplay-blocking/m-p/1950039`
   - so we have 4 users already who confirmed the issue

**Reply by Readmit0** [reply]
Any updates on fixing it?

**Reply by krystian3w** [reply]
Switch to Vivaldi?
